elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+
+            otherwise ->
+                ":)"
+
+
+describe_option : Option -> Html Msg
+describe_option option =
+    div [ class "search-details" ]
+        [ table []
+            [ tbody []
+                [ tr []
+                    [ th [] [ text "Description:" ]
+                    , td [] []
+                    , td [ class "description docbook" ] [ text (description_to_text option.description) ]
+                    ]
+                , tr []
+                    [ th [] [ text "Default Value:" ]
+                    , td [] []
+                    , td [ class "default" ] [ null_is_not_given nix option.default ]
+                    ]
+                , tr []
+                    [ th [] [ text "Example value:" ]
+                    , td [] []
+                    , td [ class "example" ] [ null_is_not_given nix option.example ]
+                    ]
+                , tr []
+                    [ th [] [ text "Declared In:" ]
+                    , td [] []
+                    , td [ class "declared-in" ] (List.map declaration_link option.declarations)
+                    ]
+                ]
+            ]
+        ]
+
+
+optionToTd : Maybe String -> ( String, Option ) -> List (Html Msg)
+optionToTd selected ( name, option ) =
+    let
+        isSelected =
+            (case selected of
+                Nothing ->
+                    False
+
+                Just selopt ->
+                    selopt == name
+            )
+    in
+        (List.concat
+            [ [ (tr []
+                    [ td
+                        [ onClick
+                            (if isSelected then
+                                (DeselectOption)
+                             else
+                                (SelectOption name)
+                            )
+                        ]
+                        [ tt [] [ text name ] ]
+                    ]
+                )
+              ]
+            , (if not isSelected then
+                []
+               else
+                [ tr []
+                    [ td [ class "details" ] [ describe_option option ]
+                    ]
+                ]
+              )
+            ]
+        )
+
+
+opt_tuple_matches_name : String -> ( String, Option ) -> Bool
+opt_tuple_matches_name term ( name, option ) =
+    term == name
+
+
+changePageIf : Bool -> Int -> String -> Html Msg
+changePageIf cond page txt =
+    button
+        (if cond then
+            [ onClick (ChangePage page) ]
+         else
+            [ disabled True ]
+        )
+        [ text txt ]
+
+
+view : Model -> Html Msg
+view model =
+    if List.length model.options > 0 then
+        viewOptions model
+    else
+        div [] [ h1 [] [ text (toString model.status) ] ]
+
+
+debounce =
+    Debounce.debounceCmd debounce_cfg
+
+
+debounce_cfg : Debounce.Config Model Msg
+debounce_cfg =
+    Debounce.config
+        .debounce_state
+        -- getState   : Model -> Debounce.State
```

Comment:
I think putting the comments on the preceding line will clarify things, and make the "trailing" comment thing go away